core/schemas/package.py

```diff
@@ -0,0 +1,276 @@
+                raise ValueError(f'Relationship source "{source}" does not exist')
+            for relationship in relationships:
+                if relationship.target not in self._objects:
+                    raise ValueError(
+                        f'Relationship target "{relationship.target}" does not exist'
+                    )
+        for observable_element in self.observables:
+            self._save_observable(observable_element)
+        for entity_element in self.entities:
+            self._save_entity(entity_element)
+        for indicator_element in self.indicators:
+            self._save_indicator(indicator_element)
+        self._save_relationships()
+
+    def _convert_timestamp(self, timestamp: str | int) -> datetime:
+        if isinstance(timestamp, int):
+            if timestamp > 10000000000:
+                return datetime.fromtimestamp(timestamp / 1000, tz=timezone.utc)
+            else:
+                return datetime.fromtimestamp(timestamp, tz=timezone.utc)
+        elif isinstance(timestamp, str):
+            if "." in timestamp:
+                fmt = "%Y-%m-%dT%H:%M:%S.%f%z"
+            else:
+                fmt = "%Y-%m-%dT%H:%M:%S%z"
+            return datetime.strptime(timestamp, fmt)
+        else:
+            raise ValueError("Invalid timestamp format")
+
+    def _save_entity(self, element: entity.EntityTypes) -> None:
+        yeti_entity = entity.Entity.find(name=element.name, type=element.type)
+        if not yeti_entity:
+            yeti_entity = element.save()
+        if hasattr(yeti_entity, "first_seen") and hasattr(yeti_entity, "last_seen"):
+            yeti_entity.first_seen = (
+                self._timestamp_dt
+                if yeti_entity.first_seen > self._timestamp_dt
+                else yeti_entity.first_seen
+            )
+            yeti_entity.last_seen = (
+                self._timestamp_dt
+                if yeti_entity.last_seen < self._timestamp_dt
+                else yeti_entity.last_seen
+            )
+            yeti_entity = yeti_entity.save()
+        if self.tags:
+            yeti_entity.tag(self.tags)
+        yeti_entity = self._update_entity_context(yeti_entity)
+        self._objects[element.name] = yeti_entity.save()
+
+    def _save_indicator(self, element: indicator.IndicatorTypes) -> None:
+        yeti_indicator = indicator.Indicator.find(name=element.name, type=element.type)
+        if not yeti_indicator:
+            yeti_indicator = element.save()
+        if self.tags:
+            yeti_indicator.tag(self.tags)
+        self._objects[element.name] = yeti_indicator.save()
+
+    def _save_observable(self, element: observable.ObservableTypes) -> None:
+        yeti_observable = observable.Observable.find(
+            value=element.value, type=element.type
+        )
+        tags = set(self.tags)
+        if not yeti_observable:
+            # support unknown observable type with generic and adds type as tag: type:<obs_type>
```

Comment:
Given that this is not the usual behavior that we have when saving objects, can we add a `tag_unknown_with_type` parameter to this `_save_observable` function to do this?

Reply:
What would be the behaviour if not enabled?

Reply:
it would not add the extra tag to the observables that are added

Reply:
And how would you imagine setting this when building from json for example?